source/uwp/Renderer/lib/ActionHelpers.cpp

```diff
@@ -0,0 +1,595 @@
+                                                   buttonText](IInspectable* /*sender*/, IRoutedEventArgs * /*args*/) -> HRESULT {
+                        ComPtr<IFrameworkElement> buttonTextAsFrameworkElement;
+                        RETURN_IF_FAILED(buttonText.As(&buttonTextAsFrameworkElement));
+
+                        return SetMatchingHeight(buttonIconAsFrameworkElement.Get(), buttonTextAsFrameworkElement.Get());
+                    }).Get(),
+                    &eventToken));
+
+                // Only add spacing when the icon must be located at the left of the title
+                UINT spacingSize;
+                THROW_IF_FAILED(GetSpacingSizeFromSpacing(hostConfig, ABI::AdaptiveNamespace::Spacing::Default, &spacingSize));
+
+                ABI::Windows::UI::Color color = {0};
+                separator = XamlHelpers::CreateSeparator(renderContext, spacingSize, spacingSize, color, false);
+            }
+
+            ComPtr<IPanel> buttonContentsPanel;
+            THROW_IF_FAILED(buttonContentsStackPanel.As(&buttonContentsPanel));
+
+            // Add image to stack panel
+            XamlHelpers::AppendXamlElementToPanel(buttonIcon.Get(), buttonContentsPanel.Get());
+
+            // Add separator to stack panel
+            if (separator != nullptr)
+            {
+                XamlHelpers::AppendXamlElementToPanel(separator.Get(), buttonContentsPanel.Get());
+            }
+
+            // Add text to stack panel
+            XamlHelpers::AppendXamlElementToPanel(buttonText.Get(), buttonContentsPanel.Get());
+
+            // Finally, put the stack panel inside the final button
+            ComPtr<IContentControl> buttonContentControl;
+            THROW_IF_FAILED(localButton.As(&buttonContentControl));
+            THROW_IF_FAILED(buttonContentControl->put_Content(buttonContentsPanel.Get()));
+        }
+        else
+        {
+            XamlHelpers::SetContent(localButton.Get(), title.Get());
+        }
+    }
+
+    HRESULT HandleActionStyling(_In_ IAdaptiveActionElement* adaptiveActionElement,
+                                _In_ IFrameworkElement* buttonFrameworkElement,
+                                _In_ IAdaptiveRenderContext* renderContext)
+    {
+        HString actionSentiment;
+        RETURN_IF_FAILED(adaptiveActionElement->get_Style(actionSentiment.GetAddressOf()));
+
+        INT32 isSentimentPositive{}, isSentimentDestructive{}, isSentimentDefault{};
+
+        ComPtr<IResourceDictionary> resourceDictionary;
+        RETURN_IF_FAILED(renderContext->get_OverrideStyles(&resourceDictionary));
+        ComPtr<IStyle> styleToApply;
+
+        ComPtr<AdaptiveNamespace::AdaptiveRenderContext> contextImpl =
+            PeekInnards<AdaptiveNamespace::AdaptiveRenderContext>(renderContext);
+
+        if ((SUCCEEDED(WindowsCompareStringOrdinal(actionSentiment.Get(), HStringReference(L"default").Get(), &isSentimentDefault)) &&
+             (isSentimentDefault == 0)) ||
```

Comment:
>  [](start = 12, length = 1)

nit: extra space

Reply:
this is how `clang-format` spits it out. it's technically correct, though, because of the extra `(` nesting

---
In reply to: [326325757](https://github.com/microsoft/AdaptiveCards/pull/3457#discussion_r326325757) [](ancestors = 326325757)